lib/spot-request-poller.js

```diff
@@ -0,0 +1,113 @@
+const assert = require('assert');
+const {runAWSRequest} = require('./aws-request');
+const Iterate = require('taskcluster-lib-iterate');
+
+const log = require('./log');
+
+const BATCH_SIZE = 100;
+
+class SpotRequestPoller {
+
+  constructor({ec2, regions, runaws = runAWSRequest, spotPollDelay = 25, state}) {
+    assert(typeof ec2 === 'object');
+    assert(Array.isArray(regions));
+    regions.forEach(region => assert(typeof region == 'string'));
+    assert(typeof state === 'object');
+
+    this.ec2 = ec2;
+    this.regions = regions;
+    this.state = state;
+    this.runaws = runaws;
+
+    this.iterator = new Iterate({
+      maxIterationTime: 1000 * 60 * 15,
+      watchDog: 1000 * 60 * 15,
+      maxFailures: 10,
+      waitTime: spotPollDelay,
+      handler: async (watchdog) => {
+        try {
+          await this.poll();
+        } catch (err) {
+          console.dir(err.stack || err);
+          monitor.reportError(err, 'Error polling pricing data');
+        }
+      }
+    });
+  };
+
+  start() {
+    return this.iterator.start();
+  }
+
+  stop() {
+    return this.iterator.stop();
+  }
+
+  async poll() {
+    await Promise.all(this.regions.map(region => this._poll(region)));
+  }
+
+  async _poll(region) {
+    let idsToPoll = await this.state.spotRequestsToPoll({region});
```

Comment:
can we do an assertion that region is a string?

Reply:
We do that [here](https://github.com/taskcluster/ec2-manager/pull/26/files/3ed978da12351bf42682993634fea1b4c641b4d2#diff-bb870db80b11d148f2a4f0445f4bbfaaR14); should we double-check it?